internal/plugin/connectors/tcp/mssql/connector_test.go

```diff
@@ -21,11 +21,6 @@ func TestSingleUseConnector_Connect(t *testing.T) {
 
```

Comment:
In the line right above this (line 20), we have:
```
    expectedBackendConn := mock.NewNetConn(nil)
```
This is allocating a new netConn object. Later, when mock.DefaultCOnnectorOptions is called, that will allocate another new netConn object. After we run connect, it asserts that these two objects are "Equal".

Would it be better to set `expectedBackendConn` to the actual value (object) we expect?:
```
expectedBackendConn := mock.DefaultMSSQLConnectorCtor.BackendConn
```
The test works, so I'm thinking assert.Equal() only checks that 2 objects have the same content.